src/ctap2/commands/get_assertion.rs

```
@@ -205,6 +320,49 @@ impl GetAssertion {
             result.extensions.app_id =
                 Some(result.assertion.auth_data.rp_id_hash == RelyingParty::from(app_id).hash());
         }
+
+        // 2. prf
+        //      If the prf extension was requested and hmac-secret returned secrets,
+        //      we need to decrypt and output them as prf client outputs.
+        match self.extensions.hmac_secret {
+            Some(HmacGetSecretOrPrf::HmacGetSecret(_)) => {
+                result.extensions.hmac_get_secret =
+                    if let Some(hmac_response @ HmacSecretResponse::Secret(_)) =
+                        &result.assertion.auth_data.extensions.hmac_secret
+                    {
+                        dev.get_shared_secret()
+                            .and_then(|shared_secret| hmac_response.decrypt_secrets(shared_secret))
+                            .and_then(Result::ok)
```

Since we might swallow an error here, it would be nice to add some `debug!()` output to the error paths in `decrypt_secrets`.

---

Done! https://github.com/mozilla/authenticator-rs/pull/337/commits/3c04ddabb23310fde06e704175822500a69d5eae